### PR TITLE
Update fake metrics using REST endpoint

### LINKED ISSUE #404: Support /fakemetrics HTTP endpoint
Support /fakemetrics HTTP endpoint:
method POST
payload JSON FakeMetrics structure 

Updates the fake metrics.

If the simulator doesn't have FakeMetrics already, return BadRequest.
Only metrics that are in the payload will be updated.

### PR DESCRIPTION
**Pull request by irar2**
Closes #404

### CODE REVIEW

File: pkg/tests/fake_metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,981 @@
...
+			Expect(err).NotTo(HaveOccurred())
+			metrics = string(data)
+
+			// After update: latency buckets [1, 0, 0, 1] → counts: 1, 1, 1, 2, 2, ...
+			for i, boundary := range common.RequestLatencyBucketsBoundaries {
+				switch {
+				case i < 1:
+					expectedCount = 1
+				case i < 3:
+					expectedCount = 1
```
Comment (mayabar) [inline pkg/tests/fake_metrics_test.go]: please consolidate 2 cases

File: pkg/tests/fake_metrics_test.go [hunk 2]
```diff
@@ -0,0 +1,981 @@
...
+			Expect(err).NotTo(HaveOccurred())
+			metrics := string(data)
+
+			Expect(metrics).To(ContainSubstring(`vllm:lora_requests_info{max_lora="1",running_lora_adapters="lora1,lora2",waiting_lora_adapters="lora3"} 1.000000001e+09`))
+			Expect(metrics).To(ContainSubstring(`vllm:lora_requests_info{max_lora="1",running_lora_adapters="lora1",waiting_lora_adapters=""} 1.000000002e+09`))
+
+			// Update lora metrics
+			reqBody := `{
+				"loras":[
+					{"running":"lora4","waiting":"lora5,lora6","timestamp":2000000001}
```
Comment (mayabar) [inline pkg/tests/fake_metrics_test.go:787]: please add test for new value as an empty array

File: pkg/tests/fake_metrics_test.go [hunk 3]
```diff
@@ -0,0 +1,981 @@
...
+
+				verifyTokenMetrics(metrics, vllmsim.PromptTokensMetricName, vllmsim.PromptTokensTotalMetricName, secondPrompt)
+				verifyTokenMetrics(metrics, vllmsim.GenerationTokensMetricName, vllmsim.GenerationTokensTotalMetricName, secondGen)
+			},
+			// Prompt tokens: hist+total → update hist → update total
+			// Generated tokens: only total → add hist → update hist
+			Entry("#1 Prompt tokens: hist+total, Generated tokens: only total; then update hists, then update totals",
+				`{"request-prompt-tokens":[1,2,3], "total-prompt-tokens":12345, "total-generation-tokens":54321}`,
+				tokenTestPhase{checkBuckets123, intPtr(12345)}, tokenTestPhase{nil, intPtr(54321)},
+				`{"request-prompt-tokens":[10, 20], "request-generation-tokens":[10, 20]}`,
```
Comment (mayabar) [inline pkg/tests/fake_metrics_test.go:920]: consider behavior change:
If histogram is defined - the total metrics value will be defined by total calculated by the histogram.
To set the total metrics - user must explicitly define it value in the json.

File: pkg/tests/metrics_test.go [hunk 4]
```diff
@@ -1290,8 +928,55 @@ var _ = Describe("estimateTokenTotal", func() {
...
 })
+
+// EstimateTokenTotal estimates the total number of tokens based on histogram bucket boundaries
+// and the number of requests in each bucket. It assumes that requests in a bucket have token
+// lengths uniformly distributed between the bucket's lower and upper bounds, and uses the
+// midpoint as a representative value for estimation.
+//
+// The last bucket is treated as [buckets[len(buckets)-1], +Inf), so its upper bound is approximated
+// as twice the lower bound for midpoint calculation.
+func estimateTokenTotal(counts []int, buckets []float64) int64 {
```
Comment (mayabar) [inline pkg/tests/metrics_test.go]: remove this function + it's usage

please move the test which uses this function to llm-d-inference-sim + test initFakeHistogram

File: pkg/llm-d-inference-sim/fake_metrics.go [hunk 5]
```diff
@@ -384,6 +389,10 @@ func (s *SimContext) UpdateFakeMetrics(fakeMetricsMap map[string]any, oldFakeMet
 		}
 	}
 
+	if generatedFakeMetricsWasEmpty && len(s.metrics.generatedFakeMetrics) > 0 {
+		go s.updateFakeMetrics()
+	}
```
Comment (mayabar) [inline pkg/llm-d-inference-sim/fake_metrics.go:402]: please stop the timer if len(s.metrics.generatedFakeMetrics)==0
